secant/Features/Scan/UIKitBridge/ScanUIView.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  ScanUIView.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 16.05.2022.
+//
+
+import Foundation
+import UIKit
+import AVFoundation
+
+public class ScanUIView: UIView {
+    var captureSession: AVCaptureSession?
+    var continueVideoSessionAfterScanning = false
+    
+    /// Rect of interest = area of the camera view used to try to recognize the qr codes.
+    private var internalRectOfInterest = CGRect(x: 0.0, y: 0.0, width: 1.0, height: 1.0)
+    var rectOfInterest: CGRect {
+        get { internalRectOfInterest }
+        set {
+            internalRectOfInterest = newValue
+            doInitialSetup()
```

Comment:
This setup seems quite expensive. Can AVCaptureSession be preserved and the rectOfInterest updated on it? SwiftU can refresh the view many times, the UIView won't be dealloc'd and recreated instead it will have its values updated. Meaning that this could be potentially updated several times in a short time and this setter be invoked more than one could think of.